sync/class.jetpack-sync-module-posts.php

```diff
@@ -140,4 +140,33 @@ public function expand_post_ids( $args ) {
 			$this->get_term_relationships( $post_ids ),
 		);
 	}
+
+	/**
+	 * Attempt to backfill up to the last 1,000 posts with $meta_key.
+	 *
+	 * Backfilling is necessary after adding meta keys to the whitelist.
+	 *
+	 * @param $meta_key
+	 *
+	 * @return array|bool
+	 */
+	public function backfill_posts_with_meta_key_config( $meta_key ) {
+		global $wpdb;
+		if ( ! $meta_key ) {
+			return false;
+		}
+
+		$post_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT post_id FROM {$wpdb->postmeta} WHERE meta_key = %s LIMIT 1000 ORDER BY post_id DESC",
```

Comment:
You might want to say "SELECT DISTINCT..." here because, although it's unlikely in our initial use case, you can have multiple meta values for the same post_id, and that would result in duplicates of the post being synced.